### ISSUE TITLE
Question about `predict_density` for MultiClass likelihood

### ISSUE THREAD

**Issue by john-bradshaw** [GPflow/GPflow#355]
Hey,

I had a question about lines 445-451 of GPflow/likelihoods.py, ie 
``` python
    def predict_density(self, Fmu, Fvar, Y):
        if isinstance(self.invlink, RobustMax):
            gh_x, gh_w = hermgauss(self.num_gauss_hermite_points)
            p = self.invlink.prob_is_largest(Y, Fmu, Fvar, gh_x, gh_w)
            return p * (1. - self.invlink.epsilon) + (1. - p) * self.invlink._eps_K1
        else:
            raise NotImplementedError
```

So as I understand it this function will be called by the GPmodel class's predict_density method, which states that it returns log probabilities. I just wanted to check that this was the case and that if so that the corresponding method of the MultiClass likelihood class did log the probabilities?
(Also as an aside this method does not affect the training though I think?)

I'm asking as I'm getting odd looking density values and also cannot work out why the `variational_expectations` method ends with `return p * np.log(1 - self.invlink.epsilon) + (1. - p) * np.log(self.invlink._eps_K1)` whereas the predict_density method ends with an almost identical statement but missing the logs: `return p * (1. - self.invlink.epsilon) + (1. - p) * self.invlink._eps_K1`.

Thanks for your help!

**Comment by alexggmatthews** [comment]
Good spot. That is a bug. Would you like to try fixing it? We should check why this isn't picked up in the unit tests also.

**Comment by alexggmatthews** [comment]
@jameshensman I'm surprised this hasn't caused trouble sooner?

**Comment by jameshensman** [comment]
Me too. Let's make sure there's a regression test. 

How about changing predict_density for predict_log_density, for clarity? With suitable deprecation warnings of course

**Comment by john-bradshaw** [comment]
So I think the `testSymmetric` method of `TestRobustMaxMulticlass` was incorrect, which was why it was passing. Some of the test methods of the `TestPredictConditional` class were also indirectly testing this method too through their call of `predict_mean_and_var`, however, I assume as the variance for the f values for these calls were 0 that the p calculated would be either almost exactly 1.0 or 0.0. And then
exp(1 * ln(1-10^-3) + 0* ...) = 1* (1 - 10^-3) + 0 * ...
(although I haven't checked for sure this is actually what is happening.)

I've created a pull request:
https://github.com/GPflow/GPflow/pull/359

Please let me know what you think. I haven't changed all the `predict_density` to `predict_log_density`, but happy to do so in a follow up PR if you want?

**Comment by alexggmatthews** [comment]
Fixed in #359
